src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineView.xaml.cs

```diff
@@ -0,0 +1,188 @@
+            _collectionView = ((CollectionViewSource)FindResource("DocumentSymbolItems")).View;
+
+            // We don't think the shell is initialized lazily, so we'll Debug.Fail(), but if it was we'd still
+            // see the view created later so this will still function.
+            if (ErrorHandler.Failed(codeWindow.GetPrimaryView(out var primaryTextView)))
+                Debug.Fail("GetPrimaryView failed during DocumentOutlineControl initialization.");
+
+            if (ErrorHandler.Failed(StartTrackingView(primaryTextView)))
+                Debug.Fail("StartTrackingView failed during DocumentOutlineControl initialization.");
+
+            if (ErrorHandler.Succeeded(codeWindow.GetSecondaryView(out var secondaryTextView)))
+            {
+                if (ErrorHandler.Failed(StartTrackingView(secondaryTextView)))
+                    Debug.Fail("StartTrackingView failed during DocumentOutlineControl initialization.");
+            }
+
+            _codeWindowEventsSink = ComEventSink.Advise<IVsCodeWindowEvents>(codeWindow, this);
+        }
+
+        private int StartTrackingView(IVsTextView textView)
+        {
+            var wpfTextView = _editorAdaptersFactoryService.GetWpfTextView(textView);
+            if (wpfTextView is null)
+                return VSConstants.E_FAIL;
+
+            _trackedTextViews.Add(textView, wpfTextView);
+
+            // In the split window case, there's two views (each with its own caret position) but only one text buffer.
+            // Subscribe to caret position changes once per view.
+            wpfTextView.Caret.PositionChanged += Caret_PositionChanged;
+
+            return VSConstants.S_OK;
+        }
+
+        private void ExpandAll(object sender, RoutedEventArgs e)
+            => _viewModel.EnqueueExpandOrCollapse(ExpansionOption.Expand);
+
+        private void CollapseAll(object sender, RoutedEventArgs e)
+            => _viewModel.EnqueueExpandOrCollapse(ExpansionOption.Collapse);
+
+        private void SearchBox_TextChanged(object sender, EventArgs e)
+        {
+            var newText = SearchBox.Text ?? string.Empty;
+            _viewModel.EnqueueFilter(newText);
+        }
+
+        private void SortByName(object sender, EventArgs e)
+            => UpdateSort(SortOption.Name);
+
+        private void SortByOrder(object sender, EventArgs e)
+            => UpdateSort(SortOption.Location);
+
+        private void SortByType(object sender, EventArgs e)
+            => UpdateSort(SortOption.Type);
+
+        private void UpdateSort(SortOption sortOption)
+        {
+            // Log which sort option was used
+            Logger.Log(sortOption switch
+            {
+                SortOption.Name => FunctionId.DocumentOutline_SortByName,
+                SortOption.Location => FunctionId.DocumentOutline_SortByOrder,
+                SortOption.Type => FunctionId.DocumentOutline_SortByType,
```

Comment:
We did separate IDs versus just a single ID w/ an extra parameter? Any idea why? Seems like this would be harder for querying.

Reply:
It's _much_ easier to query this way. You can filter on the event name directly. additional properties require you to manually split out the property bag into a temporary table and query that.